### DISCUSSION TITLE
Rearrange item in main window via drag and drop

### DISCUSSION THREAD

**Discussion by Hi-360** [1Remote/1Remote#349] [Ideas]
在日常使用中有的主机远程桌面或者ssh使用比较频繁 ，但是它的位置在最下面，每次使用的时候需要去最下面找到该主机很是麻烦 ，如果能支持该主机的图标拖动  拖动到最上面就完美了。

**Comment by majkinetor** [comment]
Not sure I understand the point. This can not be done in main window as it has sorting already, and launcher already keeps recents up. There are also tags for limiting number of connections in a view, 

To make custom order in the main connection list one would need to first somehow disable sorting and rearange stuff and in that case, its just easier to keep a special tag for it. This would require a lot of development for next to 0 benefits.

We are having again here _scratching your own itch_ type of enhacement request. I dont think this should be done, if I understand request correctly.

**Reply by xieyuqiang** [reply]
I can custom order in the main connection list by renaming the connections and sorting by name, however, the order will change back to the default one the next time I open 1Remote. So if it needs a lot of development to realize drag and drop, then a switch for stopping the order changing back to the default one would be a solution.

**Reply by VShawn** [reply]
Ahh it's a bug that `the order will change back to the default one the next time I open 1Remote`

**Comment by VShawn** [comment]
maybe some of these people dont like to use the launcher and prefer a custom order list to let them to find a server easier.

**Reply by Hi-360** [reply]
是的  非常需要这个功能

**Reply by majkinetor** [reply]
I strongly beleive that is the wrong UX. Not using collections with bunch of servers is just wrong. If you have collection of connections (via tags) then you can keep your list small. If you want something on that list to be first, just make it by name. 

In the smaller list custom order makes more sense to be honest.

**Comment by VShawn** [comment]
Found it pretty easy to make this drag & drop. 

if I can done the joy in 2h, I will add this feature  :)

**Reply by majkinetor** [reply]
The problem with this is that "main window" as concept is existent only if user has no tags. As soon as he has them, and he will, connection can appear in any number of tabs. Now you need ordering by the tag, which has next to 0 value. 

Accordingly, this shouldn't be done.
